compiler/rustc_target/src/asm/m68k.rs

```diff
@@ -0,0 +1,77 @@
+use super::{InlineAsmArch, InlineAsmType};
+use rustc_macros::HashStable_Generic;
+use rustc_span::Symbol;
+use std::fmt;
+
+def_reg_class! {
+    M68k M68kInlineAsmRegClass {
+        reg,
+        reg_ptr,
+    }
+}
+
+impl M68kInlineAsmRegClass {
+    pub fn valid_modifiers(self, _arch: super::InlineAsmArch) -> &'static [char] {
+        &[]
+    }
+
+    pub fn suggest_class(self, _arch: InlineAsmArch, _ty: InlineAsmType) -> Option<Self> {
+        None
+    }
+
+    pub fn suggest_modifier(
+        self,
+        _arch: InlineAsmArch,
+        _ty: InlineAsmType,
+    ) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn default_modifier(self, _arch: InlineAsmArch) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn supported_types(
+        self,
+        _arch: InlineAsmArch,
+    ) -> &'static [(InlineAsmType, Option<Symbol>)] {
+        match self {
+            Self::reg | Self::reg_ptr => {
+                types! { _: I8, I16, I32; }
+            }
+        }
+    }
+}
+
+def_regs! {
+    M68k M68kInlineAsmReg M68kInlineAsmRegClass {
+        d0: reg = ["d0"],
+        d1: reg = ["d1"],
+        d2: reg = ["d2"],
+        d3: reg = ["d3"],
+        d4: reg = ["d4"],
+        d5: reg = ["d5"],
+        d6: reg = ["d6"],
+        d7: reg = ["d7"],
+        a0: reg_ptr = ["a0"],
+        a1: reg_ptr = ["a1"],
+        a2: reg_ptr = ["a2"],
+        a3: reg_ptr = ["a3"],
+        a4: reg_ptr = ["a4"],
```

Comment:
`a4` is reserved as the base pointer by LLVM.

Reply:
Ah. Though it's `a5`, not `a4`, looking at https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/M68k/M68kRegisterInfo.td.

Looking at that file there are a few extra register name aliases to add here too.

Reply:
GCC seems to reserve `a5` for PIC code though, so it might make sense to disallow its use in inline asm as well: https://github.com/gcc-mirror/gcc/blob/master/gcc/config/m68k/m68k.h#L328